pkg/apb/bind.go

```diff
@@ -51,22 +51,19 @@ func Bind(
 	log.Debug("Calling getPodName")
 	podname, _ := getPodName(output, log)
 	log.Debug("Calling monitorOutput on " + podname)
-	bindout, _ := monitorOutput(podname)
-	log.Info(string(bindout))
 
-	// HACK ALERT!
-	bd, err := buildBindData(bindout)
-	if err != nil {
-		// HACK: this is HORRIBLE. but there is definitely a time between a bind
-		// and when the container is up.
-		time.Sleep(5 * time.Second)
-		log.Warning("Trying to monitor the output again on " + podname)
-		bindout, _ = monitorOutput(podname)
-		log.Info(string(bindout))
-		return buildBindData(bindout)
+	out := make(chan []byte)
+	go func() {
+		monitorOutput(podname, out, log)
+	}()
+
+	var bindout []byte
+	for msg := range out {
+		log.Info(string(msg))
+		bindout = msg
```

Comment:
So we monitor the output in a go routine, then we wait for there to be something in the channel to parse. I like it.